c-bindings/c-bindings.go

```diff
@@ -4,33 +4,27 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"unsafe"
 
 	"github.com/google/go-jsonnet"
+	"github.com/google/go-jsonnet/ast"
 
 	// #cgo CXXFLAGS: -std=c++11 -Wall -I../cpp-jsonnet/include
 	// #include "internal.h"
 	"C"
 )
 
-const maxID = 1000
-
 type vm struct {
 	*jsonnet.VM
 	importer *jsonnet.FileImporter
 }
 
-// Because of Go GC, there are restrictions on keeping Go pointers in C.
-// We cannot just pass *jsonnet.VM to C. So instead we use "handle" structs in C
-// which refer to JsonnetVM by a numeric id.
-// The way it is implemented below is simple and has low overhead, but requires us to keep
-// a list of used IDs. This results in a permanent "leak". I don't expect it to ever
-// become a problem.
-// The VM IDs start with 1, so 0 is never a valid ID and the VM's index in the array is (ID - 1).
-
-// VMs is the set of active, valid Jsonnet virtual machine handles allocated by jsonnet_make.
-var VMs = []*vm{}
-var freedIDs = []uint32{}
+type jsonValue struct {
+	obj      interface{}
```

Comment:
`obj` is not a very good name - objects are just one kind of JSON values. I would go with `val`.